environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceEncryptionCreationHandler.java

```diff
@@ -0,0 +1,89 @@
+package com.sequenceiq.environment.environment.flow.creation.handler;
+
+import static com.sequenceiq.cloudbreak.common.mappable.CloudPlatform.AZURE;
+import static com.sequenceiq.environment.environment.flow.creation.event.EnvCreationHandlerSelectors.CREATE_ENVIRONMENT_RESOURCE_ENCRYPTION_EVENT;
+import static com.sequenceiq.environment.environment.flow.creation.event.EnvCreationStateSelectors.START_FREEIPA_CREATION_EVENT;
+
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.apache.commons.lang3.StringUtils;
+
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationEvent;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationFailureEvent;
+import com.sequenceiq.environment.environment.service.EnvironmentService;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class ResourceEncryptionCreationHandler extends EventSenderAwareHandler<EnvironmentDto> {
```

Comment:
Perhaps rename `ResourceEncryptionCreationHandler` to `ResourceEncryptionInitializationHandler`.